Sources_v3/StreamChatUI/ChatChannelList/ChatUnreadCountBubbleView.swift

```diff
@@ -0,0 +1,87 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+open class ChatUnreadCountBubbleView: UIView {
+    // MARK: - Properties
+    
+    public var inset: CGFloat = 3
+    
+    override open var intrinsicContentSize: CGSize {
+        let height: CGFloat = max(unreadCountLabel.font.pointSize + inset * 2, frame.height)
+        let width = max(unreadCountLabel.intrinsicContentSize.width + inset * 2, height)
+        return .init(width: width, height: height)
+    }
+    
+    public var unreadCount: ChannelUnreadCount = .noUnread {
+        didSet {
+            updateContent()
+        }
+    }
+    
+    // MARK: - Subviews
+    
+    private let unreadCountLabel: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        return label
+    }()
+    
+    // MARK: - Init
+    
+    override public init(frame: CGRect) {
+        super.init(frame: frame)
+        commonInit()
+    }
+    
+    public required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        commonInit()
+    }
+    
+    private func commonInit() {
+        setupLayout()
+        setupAppearance()
+        updateContent()
+    }
+    
+    // MARK: - Layout
+    
+    override open func invalidateIntrinsicContentSize() {
+        super.invalidateIntrinsicContentSize()
+        
+        unreadCountLabel.invalidateIntrinsicContentSize()
+    }
+    
+    override open func layoutSubviews() {
+        super.layoutSubviews()
+        
+        invalidateIntrinsicContentSize()
```

Comment:
I don't think we should call this in `layoutSubviews`

Reply:
Without this call bubble is not changed on `invalidateLayout()` in `ChatChannelListCollectionViewLayout` after `UIContentSizeCategory.didChangeNotification`.